modules/shared/src/main/scala/org/tessellation/security/signature/Signed.scala

```diff
@@ -33,6 +33,10 @@ import org.scalacheck.Arbitrary.arbitrary
 case class Signed[+A](value: A, proofs: NonEmptySet[SignatureProof])
 
 object Signed {
+  implicit val functor: Functor[Signed] = new Functor[Signed] {
+    def map[A, B](fa: Signed[A])(f: A => B): Signed[B] = Signed(f(fa.value), fa.proofs)
```

Comment:
Shouldn't applying a function to any value invalidate the associated proofs?

Reply:
Not always. CurrencyIncrementalSnapshotV1 should be equivalent to CurrencyIncrementalSnapshot with that difference that newer fields are None (schema upgrade). Hence, the calculated hash should be equally the same so the proofs should match